DigitalOffice/Tests/ProjectServiceTests/Repositories/EditProjectByIdRepositoryTests.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Linq;
+using NUnit.Framework;
+using Microsoft.EntityFrameworkCore;
+using LT.DigitalOffice.Kernel.UnitTestLibrary;
+using LT.DigitalOffice.ProjectService.Database;
+using LT.DigitalOffice.ProjectService.Repositories;
+using LT.DigitalOffice.ProjectService.Database.Entities;
+
+namespace LT.DigitalOffice.ProjectServiceUnitTests.Repositories
+{
+    public class EditProjectByIdRepositoryTests
+    {
+        private Guid departmentId;
+
+        private DbProject existingProject;
+        private DbProject editProject;
+
+        private ProjectRepository repository;
+        private ProjectServiceDbContext dbContext;
+        private DbContextOptions<ProjectServiceDbContext> dbOptionsProjectService;
+
+        [SetUp]
+        public void SetUp()
+        {
+            dbOptionsProjectService = new DbContextOptionsBuilder<ProjectServiceDbContext>()
+                .UseInMemoryDatabase(databaseName: "ProjectServiceTest")
+                .Options;
+
+            dbContext = new ProjectServiceDbContext(dbOptionsProjectService);
+            repository = new ProjectRepository(dbContext);
+
+            departmentId = Guid.NewGuid();
+
+            existingProject = new DbProject
+            {
+                Name = "A test name",
+                DepartmentId = departmentId,
+                Description = "Description",
+                IsActive = true,
+                Deferred = false
+            };
+
+            editProject = new DbProject
+            {
+                Name = "Is different",
+                DepartmentId = Guid.NewGuid(),
+                Description = "Is different too",
+                IsActive = false,
+                Deferred = false
+            };
+        }
+
+        [Test]
+        public void SuccessfulEditProjectTest()
+        {
+            existingProject.Id = Guid.NewGuid();
+            editProject.Id = existingProject.Id;
+
+            dbContext.Add(existingProject);
+            dbContext.SaveChanges();
+            dbContext.Entry(existingProject).State = EntityState.Detached;
+            dbContext.SaveChanges();
+
+            var dbProject = dbContext.Projects
+                .AsNoTracking()
+                .SingleOrDefault(p => p.Id == existingProject.Id);
+            Assert.IsNotNull(dbProject);
+
+            repository.EditProjectById(editProject.Id, editProject);
+            dbContext.Entry(editProject).State = EntityState.Detached;
+            dbContext.SaveChanges();
+
+            var editedProject = dbContext.Projects
+                .AsNoTracking()
+                .SingleOrDefault(p => p.Id == editProject.Id);
+            Assert.IsNotNull(editedProject);
+
+            Assert.AreEqual(dbProject.Id, editedProject.Id);
+            SerializerAssert.AreNotEqual(dbProject, editedProject);
```

Comment:
It makes more sense to check that editedProject is equal to editProject

Reply:
Disagree.
Well, kinda.
I'll just add one more check, but keep the first one. We should know that any update appeared in the database.

Reply:
First one is good. I said about AreNotEqual check